### PR TITLE
EPMRPP-52574: [AGENT] Jasmine. Fix hook start time

### PR DESCRIPTION
**Pull request by KatsiarynaTatarynovich**

### CODE REVIEW

File: lib/jasmine-reportportal-reporter.js [hunk 1]
```diff
@@ -216,17 +215,20 @@ class ReportportalReporter {
         let type = this.getTopLevelType();
 
         return promise.then((codeRef) => {
-            let suiteObj = this.client.startTestItem(Object.assign({
+            this.startTime = this.getTime();
```
Comment (AmsterGet) [inline lib/jasmine-reportportal-reporter.js:218]: Could you please indicate what time you hold in the class field `startTime`?

File: lib/jasmine-reportportal-reporter.js [hunk 2]
```diff
@@ -55,32 +55,32 @@ class ReportportalReporter {
...
 
-    setParentId(id) {
-        this.parentIds.push(id);
+    setParentInfo(info) {
+        this.parentInfo.push(info);
     }
 
     finishParent() {
-        this.parentIds.pop();
+        this.parentInfo.pop();
```
Comment (AmsterGet) [inline lib/jasmine-reportportal-reporter.js:79]: May be `parentsInfo`?

File: lib/jasmine-reportportal-reporter.js [hunk 3]
```diff
@@ -239,29 +241,39 @@ class ReportportalReporter {
...
 
-            this.setParentId(stepObj.tempId);
+            this.setParentInfo({ tempId: stepObj.tempId, startTime: this.startTime });
             promiseErrorHandler(stepObj.promise);
         });
     }
 
+    getHookStartTime(hookType, parent) {
+        if (hookType === entityType.BEFORE_METHOD || hookType === entityType.BEFORE_SUITE) {
+            return Math.max(parent && parent.startTime, this.startTime - 1);
```
Comment (AmsterGet) [inline lib/jasmine-reportportal-reporter.js:263]: Why `max`? Should we use the earliest time to report `before` hooks?

File: lib/jasmine-reportportal-reporter.js [hunk 4]
```diff
@@ -239,29 +241,39 @@ class ReportportalReporter {
...
+        }
+        return this.startTime || this.getTime();
```
Comment (AmsterGet) [inline lib/jasmine-reportportal-reporter.js:265]: I think that for `after` hooks we can return just `this.getTime()`. @KatsiarynaTatarynovich What do you think?

Comment (KatsiarynaTatarynovich) [reply lib/jasmine-reportportal-reporter.js:265]: I think using startTime creates a more streamlined system. If we use `this.getTime()`, the after hook may be mixed with before hook of the next test.

File: lib/jasmine-reportportal-reporter.js (hunk 3 above)
Comment (KatsiarynaTatarynovich) [reply lib/jasmine-reportportal-reporter.js:263]: This is all due to the specific work of hooks and at the same time, the speed of the tests themselves. We need max as the value of `parent.startTime` often less than `this.startTime - 1`. The value of parentStartTime is not that we need. But it helps us avoid problems if there is no `this.startTime`.

**Pull request merged**